token_info_unit_test.go

```diff
@@ -0,0 +1,143 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTokenInfo_Protobuf(t *testing.T) {
+	tokenInfo := setupTokenInfo()
+	pb := tokenInfo._ToProtobuf()
+	expected := _TokenInfoFromProtobuf(pb)
+
+	assertTokenInfo(t, tokenInfo, expected)
+}
+
+func TestTokenInfo_Bytes(t *testing.T) {
+	tokenInfo := setupTokenInfo()
+	pb := tokenInfo.ToBytes()
+	expected, _ := TokenInfoFromBytes(pb)
+
+	assertTokenInfo(t, tokenInfo, expected)
+}
+
+func TestTokenInfo_ProtobufCoverage(t *testing.T) {
+	tokenInfo := setupTokenInfo()
+
+	_true := true
+	_false := false
+
+	tokenInfo.DefaultKycStatus = &_false
+	tokenInfo.DefaultFreezeStatus = &_false
+	tokenInfo.PauseStatus = &_true
+
+	pb := tokenInfo._ToProtobuf()
+	expected := _TokenInfoFromProtobuf(pb)
+
+	assertTokenInfo(t, tokenInfo, expected)
+}
+
+func setupTokenInfo() TokenInfo {
+	publickKey, _ := PublicKeyFromStringEd25519("302a300506032b6570032100996b8090805be647ff07399e30ab1423fd3fcd795410df3c344fcd98ba27122e")
+	accId, _ := AccountIDFromString("0.0.1111")
+
+	_true := true
+	_false := false
+	ledgerId := NewLedgerIDTestnet()
+	timeDuration := time.Duration(2230000) * time.Second
+
+	// time.Time
+	timeTime := time.Unix(1230000, 0)
+	tokenId, _ := TokenIDFromString("0.0.123")
+	feeCollectorAccountId, _ := AccountIDFromString("0.0.123")
+
+	customFees := []Fee{
+		NewCustomFixedFee().
+			SetAmount(1).
+			SetDenominatingTokenID(tokenId).
+			SetFeeCollectorAccountID(feeCollectorAccountId),
+	}
+
+	return TokenInfo{
+		TokenID:             tokenId,
+		Name:                "test",
+		Symbol:              "test",
+		Decimals:            8,
+		TotalSupply:         10000,
+		Treasury:            accId,
+		AdminKey:            publickKey,
+		KycKey:              publickKey,
+		FreezeKey:           publickKey,
+		WipeKey:             publickKey,
+		SupplyKey:           publickKey,
+		DefaultFreezeStatus: &_true,
+		DefaultKycStatus:    &_true,
+		Deleted:             false,
+		AutoRenewPeriod:     &timeDuration,
+		AutoRenewAccountID:  AccountID{},
+		ExpirationTime:      &timeTime,
+		TokenMemo:           "test-memo",
+		TokenType:           TokenTypeFungibleCommon,
+		SupplyType:          TokenSupplyTypeInfinite,
+		MaxSupply:           10000000,
+		FeeScheduleKey:      publickKey,
+		CustomFees:          customFees,
+		PauseKey:            publickKey,
+		PauseStatus:         &_false,
+		LedgerID:            *ledgerId,
+	}
+}
+
+func assertTokenInfo(t assert.TestingT, tokenInfo TokenInfo, expected TokenInfo) {
+	assert.Equal(t, tokenInfo.TokenID, expected.TokenID)
+	assert.Equal(t, tokenInfo.Name, expected.Name)
+	assert.Equal(t, tokenInfo.Symbol, expected.Symbol)
+	assert.Equal(t, tokenInfo.Decimals, expected.Decimals)
+	assert.Equal(t, tokenInfo.TotalSupply, expected.TotalSupply)
+	// assert.Equal(t, tokenInfo.Treasury, expected.Treasury)
```

Comment:
After protobuf conversion `tokenInfo.Treasury` are not equal because of the nil, and `omitempty` values.

Reply:
This was actually a bug, fixed in 742